### PR TITLE
docs: Review API docs

### PR DESCRIPTION
**Pull request by lorensr**
## What was changed

- Adds links to docs & SDK docs on index
- Review 
- Excludes anything tagged `@internal`

## Next PR

- Add otel and testing to meta
- Finish review

### CODE REVIEW

File: packages/activity/src/index.ts [hunk 1]
```diff
@@ -120,9 +147,14 @@ export interface Info {
 }
 
 /**
- * Activity Context manager.
+ * Activity Context, used to:
+ *
+ * - Get {@link Info} about the current Activity Execution
+ * - Send [heartbeats](https://docs.temporal.io/concepts/what-is-an-activity-heartbeat)
+ * - Get notified of Activity cancellation
+ * - Sleep
```
Comment (bergundy) [inline packages/activity/src/index.ts:155]: I'd note that this is cancellation aware sleep

File: packages/internal-workflow-common/src/errors.ts [hunk 2]
```diff
@@ -36,11 +40,11 @@ export class WorkflowExecutionAlreadyStartedError extends Error {
 }
 
 /**
- * Thrown when workflow with the given id is not known to the Temporal service.
+ * Thrown when a Workflow with the given Id is not known to Temporal Server.
  * It could be because:
- * - ID passed is incorrect
- * - Workflow execution is complete (for some calls e.g. terminate),
- * - workflow was purged from the service after reaching its retention limit.
+ * - Id passed is incorrect
```
Comment (bergundy) [inline packages/internal-workflow-common/src/errors.ts:45]: `ID` is clearer

File: packages/internal-workflow-common/src/errors.ts [hunk 3]
```diff
@@ -67,7 +71,7 @@ interface ErrorWithCode {
   code: string;
 }
 /**
- * Get error code from an Error or return undefined
+ * * Get `error.code` (or `undefined` if not present)
```
Comment (bergundy) [inline packages/internal-workflow-common/src/errors.ts:74]:
```suggestion
 * Get `error.code` (or `undefined` if not present)
```

File: packages/workflow/src/workflow.ts [hunk 4]
```diff
@@ -915,24 +915,21 @@ export function makeContinueAsNewFunc<F extends Workflow>(
 }
 
 /**
- * Continues current Workflow execution as new with default options.
+ * [Continues-As-New](https://docs.temporal.io/concepts/what-is-continue-as-new/) the current Workflow Execution with
+ * default options.
  *
- * Shorthand for `makeContinueAsNewFunc<F>()(...args)`.
+ * Shorthand for {@link makeContinueAsNewFunc}.
```
Comment (bergundy) [inline packages/workflow/src/workflow.ts:921]: I liked the extra details.

Comment (bergundy) [reply packages/workflow/src/workflow.ts:921]: You might want to leave as it was and add a `see also {@link make...}`

File: packages/workflow/src/workflow.ts [hunk 5]
```diff
@@ -915,24 +915,21 @@ export function makeContinueAsNewFunc<F extends Workflow>(
...
  *
  * @example
  *
- * ```ts
- * import { continueAsNew } from '@temporalio/workflow';
+ *```ts
+ *import { continueAsNew } from '@temporalio/workflow';
  *
- * export function myWorkflow(n: number) {
```
Comment (bergundy) [inline packages/workflow/src/workflow.ts:927]: snipsync would have found this (but the example wouldn't show up in IDEs).
There's probably a better solution for testing the examples but it's not critical for this PR.

**Review by bergundy — APPROVED** [review]
Awesome stuff @lorensr!

All of theslinks that are dependent on content being rendered in docusaurus should change to `{@link ...}` so the docs can be used from IDEs.

File: packages/internal-workflow-common/src/errors.ts (hunk 2 above)
Comment (lorensr) [reply packages/internal-workflow-common/src/errors.ts:45]: That's my pref as well, but I was trying to go by this, which says either Id or identifier:

https://github.com/temporalio/documentation/blob/master/STYLE.md#abbreviation-of-identifier

**Comment by lorensr** [comment]
Omg this is the most useful example 😆

image

**Pull request merged**